sentry/src/routes/channel.rs

```diff
@@ -498,6 +504,68 @@ pub async fn channel_payout<C: Locked + 'static>(
     Ok(Json(SuccessResponse { success: true }))
 }
 
+/// GET `/v5/channel/0xXXX.../get-leaf/spender` request and
+/// GET `/v5/channel/0xXXX.../get-leaf/earner` request
+///
+/// Response: [`GetLeafResponse`]
+pub async fn get_leaf<C: Locked + 'static>(
+    Extension(app): Extension<Arc<Application<C>>>,
+    Extension(channel_context): Extension<ChainOf<Channel>>,
+    Extension(leaf_for): Extension<LeafFor>,
+    Path(params): Path<(ChannelId, Address)>,
+) -> Result<Json<GetLeafResponse>, ResponseError> {
+    let channel = channel_context.context;
+
+    let approve_state = match latest_approve_state(&app.pool, &channel).await? {
+        Some(approve_state) => approve_state,
+        None => {
+            return Err(ResponseError::BadRequest(
+                "No ApproveState message for spender".to_string(),
+            ))
```

Comment:
Could you please use 404 in this case. A channel can be new and might not have an ApproveState yet. This shouldn't be a `BadRequest` (400) error.